src/main/java/com/thealgorithms/datastructures/trees/SplayTree.java

```diff
@@ -0,0 +1,306 @@
+     * @return The new root of the splayed subtree.
+     */
+    private Node splay(Node root, int key) {
+        if (root == null || root.key == key) {
+            return root;
+        }
+
+        if (root.key > key) {
+            if (root.left == null) {
+                return root;
+            }
+            // Zig-Zig case
+            if (root.left.key > key) {
+                // Recursive call to splay on grandchild
+                root.left.left = splay(root.left.left, key);
+                // Perform zig operation on parent
+                root = rotateRight(root);
+            } // Zig-Zag case
+            else if (root.left.key < key) {
+                root.left.right = splay(root.left.right, key);
+                // Perform zag operation on parent
+                if (root.left.right != null) {
+                    root.left = rotateLeft(root.left);
+                }
+            }
+            return (root.left == null) ? root : rotateRight(root);
+        } else {
+            if (root.right == null) {
+                return root;
+            }
+            // Zag-Zag case
+            if (root.right.key > key) {
+                root.right.left = splay(root.right.left, key);
+                // Perform zig operation on parent
+                if (root.right.left != null) {
+                    root.right = rotateRight(root.right);
+                }
+            } // Zag-Zig case
+            else if (root.right.key < key) {
+                root.right.right = splay(root.right.right, key);
+                // Perform zag operation on parent
+                root = rotateLeft(root);
+            }
+            return (root.right == null) ? root : rotateLeft(root);
+        }
+    }
+
+    /**
+     * Insert a key into the SplayTree.
+     *
+     * @param key The key to insert.
+     */
+    public void insert(int key) {
+        root = insertRec(root, key);
+        root = splay(root, key);
+    }
+
+    /**
+     * Recursive function to insert a key into a subtree.
+     *
+     * @param root The root of the subtree to insert the key into.
+     * @param key  The key to insert.
+     * @return The root of the modified subtree after insertion.
+     * @throws IllegalArgumentException If the key to be inserted already exists in the subtree.
+     */
+    private Node insertRec(Node root, int key) {
+        if (root == null) {
+            return new Node(key);
+        }
+
+        if (key < root.key) {
+            root.left = insertRec(root.left, key);
+        } else if (key > root.key) {
+            root.right = insertRec(root.right, key);
+        } else {
+            throw new IllegalArgumentException("Duplicate key: " + key);
+        }
+
+        return root;
+    }
+
+    /**
+     * Search for a key in the SplayTree.
+     *
+     * @param key The key to search for.
+     * @return True if the key is found, otherwise false.
+     */
+    public boolean search(int key) {
```

Comment:
What do you think about methods reordering? Because usually standard ordering is assumed such order:

1. Class (static) variables: First the public class variables, then the protected, and then the private.

2. Instance variables: First public, then protected, and then private.

3. Constructors

4. Methods: should be grouped by functionality rather than by scope or accessibility. For example, a private class method can be in between two public instance methods. The goal is to make reading and understanding the code easier.

https://www.oracle.com/java/technologies/javase/codeconventions-fileorganization.html

Reply:
These best practices should be followed.